src/cryptsetup/cryptsetup.c

```
@@ -532,6 +535,9 @@ static int get_password(
         assert(src);
         assert(ret);
 
+        if (arg_headless)
+                return log_error_errno(SYNTHETIC_ERRNO(EPERM), "Password querying disabled via 'headless' option.");
```

we probably should return a common error for this. In the fido2 case you return EUNATCH, but you could also use ENOMEDIUM? ENOPKG? something that isn't use otherwise by the cryptsetup/luks/kernel/pkc11/fido2 stuff here so far...